streams/src/commonTest/kotlin/com/mirego/trikot/streams/reactive/processors/SampleProcessorTests.kt

```diff
@@ -0,0 +1,127 @@
+package com.mirego.trikot.streams.reactive.processors
+
+import com.mirego.trikot.streams.cancellable.CancellableManager
+import com.mirego.trikot.streams.reactive.Publishers
+import com.mirego.trikot.streams.reactive.sample
+import com.mirego.trikot.streams.reactive.subscribe
+import com.mirego.trikot.streams.utils.MockTimer
+import com.mirego.trikot.streams.utils.MockTimerFactory
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.time.seconds
+
+class SampleProcessorTests {
```

Comment:
We might need to unit tests to clarify the expected behaviour when there are multiple subscribers, and when the processor goes from 0 to 1 subscriber, then back to 0.

Reply:
I would also add a unit test as to what is expected to happen when the subscription is cancelled before the timer is fired.

Reply:
I added the test for the canceled subscription. But I felt like `givenAPublisherThatDoesNotEmitValueWithinAnIntervalAndASubscriptionIsMadeAfterWhenSampleThenItReceivesTheLastValue` described well enough the behavior of many subscription.